src/lang/std/engineConnection.ts

```diff
@@ -832,9 +832,9 @@ export class EngineCommandManager {
     } else {
       this.artifactMap[id] = {
         type: 'failed',
-        range: command.range,
-        commandType: command.commandType,
-        parentId: command.parentId ? command.parentId : undefined,
+        range: command !== undefined ? command.range : undefined,
```

Comment:
I am _not_ a Typescript expert, but is the same as 
```suggestion
        range: command?.range
```